api/src/repositories/attachment-repository.ts

```diff
@@ -224,6 +282,56 @@ export class AttachmentRepository extends BaseRepository {
     return response.rows[0];
   }
 
+  /**
+   * Query to return the report attachments having the given IDs and belonging to the given project.
+   * @param {number} projectId the ID of the project
+   * @param {number[]} reportAttachmentIds the IDs of the report attachments
+   * @return {Promise<IProjectReportAttachment[]>} Promise resolving the report attachment
+   * @memberof AttachmentRepository
+   */
+  async getProjectReportAttachmentsByIds(
+    projectId: number,
+    reportAttachmentIds: number[]
+  ): Promise<IProjectReportAttachment[]> {
+    defaultLog.debug({ label: 'getProjectReportAttachmentsByIds' });
+
+    const knex = getKnex();
+    const queryBuilder = knex
+      .queryBuilder()
+      .select([
+        'project_report_attachment_id as id',
+        'uuid',
+        'file_name',
+        'title',
+        'description',
+        knex.raw('year::int as year_published'),
+        knex.raw(`
+          CASE
+            WHEN update_date IS NULL
+            THEN create_date::text
+            ELSE update_date::text
+          END AS last_modified
+        `),
```

Comment:
Just curious: looks like we only do this for project/survey reports, but not attachments. Any reason? Or just do work with existing interfaces that were designed differently, or something?